src/conv-core/cmishmem.h

```diff
@@ -33,28 +33,18 @@ constexpr auto defaultTimeout = 4;
 }  // namespace ipc
 }  // namespace cmi
 
-// TODO ( find better names than src/dst? )
-#define CMK_IPC_BLOCK_FIELDS \
-  int src;                   \
-  std::uintptr_t orig;       \
-  int dst;                   \
-  std::uintptr_t next;       \
-  std::size_t size;
-
-struct CmiIpcBlock {
+struct alignas(ALIGN_BYTES) CmiIpcBlock {
+  // TODO ( find better names than src/dst? )
+public:
   // "home" rank of the block
- private:
-  class blockSizeHelper_ {
-    CMK_IPC_BLOCK_FIELDS;
-  };
-
- public:
-  CMK_IPC_BLOCK_FIELDS;
+  int src;
+  std::uintptr_t orig;
+  int dst;
+  std::uintptr_t next;
+  std::size_t size;
 
   CmiIpcBlock(std::size_t size_, std::uintptr_t orig_)
       : orig(orig_), next(cmi::ipc::nil), size(size_) {}
-
-  char padding[(sizeof(blockSizeHelper_) % ALIGN_BYTES)];
```

Comment:
Last I checked, CMI-SHMEM wasn't regularly tested. Did you ensure it still works after this change?

Reply:
No, I did not, I'll test it before I un-draft it.

But I think the original implementation is wrong in the general case, e.g. if `sizeof(blockSizeHelper_) = 10` and `ALIGN_BYTES = 8`,  `sizeof(blockSizeHelper_) % ALIGN_BYTES = 2`, yielding a final size of 12 after padding rather than 16 (separately, using `alignas` also ensures that the alignment requirement (i.e. result of `alignof`) is set correctly). In any case, I think the way it's written right now will never actually cause a problem in practice, as every existing field will end up with a padded size of 8 on every reasonable architecture, but this cleans it up and makes the struct resilient to future modifications of the fields or their order.

Reply:
Also, while you're here, can you explain the `// "home" rank of the block` comment? It seems not to relate to anything immediately around it, so I was confused as to what it was supposed to refer to.

Reply:
Ah, I see. Yeah, that conclusion seems reasonable.

The comment was likely intended to be on `src` but not moved when the rest of the fields were placed in the struct.

Reply:
By the way, another instanceof this pattern is present in [`envelope.h`](https://github.com/UIUC-PPL/charm/blob/331c13afaa2a97851757c7a522c0e863a7cedfef/src/ck-core/envelope.h#L232). That's loosely where I copied the logic for this from, but I should've just used its macros or `CMIPADDING` instead.

Reply:
> By the way, another instanceof this pattern is present in [`envelope.h`](https://github.com/UIUC-PPL/charm/blob/331c13afaa2a97851757c7a522c0e863a7cedfef/src/ck-core/envelope.h#L232). That's loosely where I copied the logic for this from, but I should've just used its macros or `CMIPADDING` instead sweat_smile

Aha, my grep of variables called `padding` didn't catch that one, thanks for pointing it out, I'll update that here as well.